wafer/schedule/views.py

```diff
@@ -203,3 +210,70 @@ def get_context_data(self, **kwargs):
                     # Must overlap with current slot
                     item['note'] = 'current'
         return context
+
+
+class ScheduleItemViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows groups to be viewed or edited.
+    """
+    queryset = ScheduleItem.objects.all()
+    serializer_class = ScheduleItemSerializer
+
+
+class ScheduleEditView(TemplateView):
+    template_name = 'wafer.schedule/edit_schedule.html'
+    date_field = 'date'
+    allow_future = True
+
+    def get_context_data(self, id=None, **kwargs):
+        context = super(ScheduleEditView, self).get_context_data(**kwargs)
+
+        days = Day.objects.all()
+
+        day = days.first()
+        if id:
+            day = days.get(id=id)
+
+        accepted_talks = Talk.objects.filter(status=ACCEPTED)
+        venues = Venue.objects.filter(days__in=[day])
+        slots = Slot.objects.filter(day=day).select_related(
+            'day', 'previous_slot').prefetch_related(
+            'scheduleitem_set', 'slot_set')
+        aggregated_slots = []
+        for slot in slots:
+            aggregated_slot = {
+                'name': slot.name,
+                'start_time': slot.get_start_time(),
+                'end_time': slot.end_time,
+                'id': slot.id,
+                'venues': []
+            }
+            for venue in venues:
+                aggregated_venue = {
+                    'name': venue.name,
+                    'id': venue.id,
+                }
+                for schedule_item in slot.scheduleitem_set.all():
+                    if schedule_item.venue.name == venue.name:
+                        if schedule_item.talk:
+                            talk = schedule_item.talk
+                            aggregated_venue['title'] = talk.title
+                            aggregated_venue['talk'] = talk
+                            aggregated_venue['scheduleitem_id'] = talk.talk_id
+                        if schedule_item.page:
+                            page = schedule_item.page
+                            aggregated_venue['title'] = page.name
+                            aggregated_venue['page'] = page
+                            aggregated_venue['scheduleitem_id'] = page.id
+
```

Comment:
Resolved in https://github.com/CTPUG/wafer/commit/57230b99f9a6028f41700a22036e8bcdd09f2d7e